rt/transports/http/src/main/java/org/apache/cxf/transport/http/HttpClientHTTPConduit.java

```diff
@@ -127,62 +128,95 @@ private static final class RefCount<T extends HttpClient> {
             this.finalizer = finalizer;
         }
 
-        RefCount<T> acquire() {
-            count.incrementAndGet();
+        /**
+         * Acquire a reference. Throws if shutdown has begun.
+         */
+        synchronized RefCount<T> acquire() {
+            if (shuttingDown) {
```

Comment:
```
   RefCount<T> acquire() {
            while (true) {
                final long c = count.get();
                if (c == 0L) {
                    throw new IllegalStateException("The client is already shutdown");
                } else if (count.compareAndSet(c, c + 1)) {
                    break;
                }
            }
            return this;
        }
```

Reply:
I think we could have a simple fix for (unless I am missing something), do you mind if I push a change ? Thank you.

Reply:
Sure, go ahead please @reta !